src/Illuminate/Validation/Concerns/FormatsMessages.php

```diff
@@ -218,15 +219,13 @@ protected function getAttributeType($attribute)
         // We assume that the attributes present in the file array are files so that
         // means that if the attribute does not have a numeric rule and the files
         // list doesn't have it we'll just consider it a string by elimination.
-        if ($this->hasRule($attribute, $this->numericRules)) {
-            return 'numeric';
-        } elseif ($this->hasRule($attribute, ['Array'])) {
-            return 'array';
-        } elseif ($this->getValue($attribute) instanceof UploadedFile) {
-            return 'file';
-        }
-
-        return 'string';
+        return match (true) {
+            $this->hasRule($attribute, $this->numericRules) => 'numeric',
+            $this->hasRule($attribute, ['Array']) => 'array',
+            $this->getValue($attribute) instanceof UploadedFile => 'file',
+            $this->getValue($attribute) instanceof File => 'file',
```

Comment:
I think a check for `instanceof Symfony\Component\HttpFoundation\File\File` would fit better here, as the ValidatesAttributes trait also uses this exact class to check against. This would also remove the need to check against both classes, as both are simply extending the `Symfony\Component\HttpFoundation\File\File` class.

```suggestion
            $this->getValue($attribute) instanceof File => 'file',
```

Reply:
Yeah, that would be a good idea I think 👍

Reply:
Agreed. Updated on https://github.com/laravel/framework/pull/49879/commits/9d741065a81668db13d43b256c7f76ed916f48ea